src/main/java/io/neonbee/internal/cluster/ClusterHelper.java

```diff
@@ -0,0 +1,107 @@
+package io.neonbee.internal.cluster;
+
+import java.util.Optional;
+import java.util.Set;
+
+import org.infinispan.manager.EmbeddedCacheManager;
+
+import com.hazelcast.cluster.Member;
+
+import io.vertx.core.Vertx;
+import io.vertx.core.impl.VertxInternal;
+import io.vertx.core.spi.cluster.ClusterManager;
+import io.vertx.ext.cluster.infinispan.InfinispanClusterManager;
+import io.vertx.spi.cluster.hazelcast.HazelcastClusterManager;
+
+public class ClusterHelper {
+
+    /**
+     * Returns the {@link ClusterManager} if NeonBee is started in clustering mode, otherwise it returns null.
+     *
+     * @param vertx {@link Vertx} instance
+     * @return null or the {@link HazelcastClusterManager} instance
+     */
+    public static Optional<ClusterManager> getClusterManager(Vertx vertx) {
+        if (vertx instanceof VertxInternal) {
+            VertxInternal vertxInternal = (VertxInternal) vertx;
+            return Optional.ofNullable(vertxInternal.getClusterManager());
+        }
+        return Optional.empty();
+    }
+
+    /**
+     * Returns an optinal with {@link HazelcastClusterManager} if NeonBee was started in cluster mode and Hazelcast is
+     * used as ClusterManager, otherwise it returns null.
+     *
+     * @param vertx {@link Vertx} instance
+     * @return null or the {@link HazelcastClusterManager} instance
+     */
+    public static Optional<HazelcastClusterManager> getHazelcastClusterManager(Vertx vertx) {
+        return getClusterManager(vertx).map(cm -> {
+            if (cm instanceof HazelcastClusterManager) {
+                return (HazelcastClusterManager) cm;
+            } else {
+                return null;
+            }
+        });
+    }
+
+    /**
+     * Returns an optinal with {@link InfinispanClusterManager} if NeonBee was started in cluster mode and Hazelcast is
+     * used as ClusterManager, otherwise it returns null.
+     *
+     * @param vertx {@link Vertx} instance
+     * @return null or the {@link InfinispanClusterManager} instance
+     */
+    public static Optional<InfinispanClusterManager> getInfinispanClusterManager(Vertx vertx) {
+        return getClusterManager(vertx).map(cm -> {
+            if (cm instanceof InfinispanClusterManager) {
+                return (InfinispanClusterManager) cm;
+            } else {
+                return null;
+            }
+        });
+    }
+
+    /**
+     * Get the cluster node ID.
+     *
+     * @param vertx {@link Vertx} instance
+     * @return the hazelcast cluster node ID
+     */
+    public static String getNodeId(Vertx vertx) {
+        return getClusterManager(vertx).map(ClusterManager::getNodeId).orElseThrow(() -> new IllegalStateException(
```

Comment:
Although I think this could be cached for Hazlecast or Infinispan, I can't make a statement about other ClusterManagers. But if it can and should be cached, for me the right place for it would be the implementation of the ClusterManager and not in NeonBee.

Reply:
Okay, as long as, `getClusterManager(vertx).map(ClusterManager::getNodeId)` is quick, it is fine for me to not cache the NodeId directly.